sklearn/preprocessing/tests/test_data.py

```diff
@@ -2654,3 +2657,113 @@ def test_kernel_centerer_feature_names_out():
     names_out = centerer.get_feature_names_out()
     samples_out2 = X_pairwise.shape[1]
     assert_array_equal(names_out, [f"kernelcenterer{i}" for i in range(samples_out2)])
+
+
+@skip_if_array_api_compat_not_configured
+@pytest.mark.parametrize("array_namespace", ["numpy.array_api", "cupy.array_api"])
+def test_minmaxscaler_array_api(array_namespace):
+    """Check that the array_api Array gives the same results as ndarrays."""
+    xp = pytest.importorskip(array_namespace)
+
+    X2d_xp = xp.asarray(X_2d)
+
+    minmax = MinMaxScaler()
+    minmax.fit(X_2d)
+
+    array_attributes = {
+        key: value
+        for key, value in vars(minmax).items()
+        if isinstance(value, np.ndarray)
+    }
+
+    minmax_xp = clone(minmax)
+    with config_context(array_api_dispatch=True):
+        minmax_xp.fit(X2d_xp)
+
+    # Fitted attributes which are arrays must have the same
+    # namespace as the one of the training data.
+    for key, attribute in array_attributes.items():
+        minmax_xp_param = getattr(minmax_xp, key)
+        assert hasattr(minmax_xp_param, "__array_namespace__")
+
+        minmax_xp_param_np = _convert_to_numpy(minmax_xp_param, xp=xp)
+        assert_allclose(
+            attribute, minmax_xp_param_np, err_msg=f"{key} not the same", atol=1e-3
+        )
+
+    # Check transformations are the same
+    methods = (
+        "transform",
+        "inverse_transform",
+    )
+
+    for method in methods:
+        result = getattr(minmax, method)(X_2d)
+        with config_context(array_api_dispatch=True):
+            result_xp = getattr(minmax_xp, method)(X2d_xp)
+        assert hasattr(
+            result_xp, "__array_namespace__"
+        ), f"{method} did not output an array_namespace"
+
+        result_xp_np = _convert_to_numpy(result_xp, xp=xp)
+
+        assert_allclose(
+            result,
+            result_xp_np,
+            err_msg=f"{method} did not the return the same result",
+            atol=1e-5,
+        )
+
+
+@skip_if_array_api_compat_not_configured
+@pytest.mark.parametrize("device", ["cuda", "cpu"])
+@pytest.mark.parametrize("dtype", ["float16", "float32", "float64"])
+def test_minmax_array_torch(device, dtype):
```

Comment:
I think we need a generic way to parametrize such tests with a list of array api libraries that we want to officially support. For now I would always test with cupy and pytorch by default (skipped when not installed).

We need some pytest helpers factorize the boiler plate, in particular because some namespace containers support multiple device options and other don't.